python_modules/libraries/dagster-fivetran/dagster_fivetran/v2/translator.py

```diff
@@ -0,0 +1,81 @@
+from enum import Enum
+from typing import Any, Mapping, Sequence
+
+from dagster import _check as check
+from dagster._core.definitions.asset_key import AssetKey
+from dagster._core.definitions.asset_spec import AssetSpec
+from dagster._record import record
+
+
+class FivetranContentType(Enum):
+    """Enum representing each object in Fivetran's ontology."""
+
+    CONNECTOR = "connector"
+    DESTINATION = "destination"
+
+
+@record
+class FivetranContentData:
+    """A record representing a piece of content in a Fivetran workspace.
+    Includes the object's type and data as returned from the API.
+    """
+
+    content_type: FivetranContentType
+    properties: Mapping[str, Any]
+
+    def to_cached_data(self) -> Mapping[str, Any]:
+        return {"content_type": self.content_type.value, "properties": self.properties}
+
+    @classmethod
+    def from_cached_data(cls, data: Mapping[Any, Any]) -> "FivetranContentData":
+        return cls(
+            content_type=FivetranContentType(data["content_type"]),
+            properties=data["properties"],
+        )
+
+
+@record
+class FivetranWorkspaceData:
+    """A record representing all content in a Fivetran workspace.
+    Provided as context for the translator so that it can resolve dependencies between content.
+    """
+
+    connectors_by_id: Mapping[str, FivetranContentData]
+    destinations_by_id: Mapping[str, FivetranContentData]
+
+    @classmethod
+    def from_content_data(
+        cls, content_data: Sequence[FivetranContentData]
+    ) -> "FivetranWorkspaceData":
+        raise NotImplementedError()
+
+
+class DagsterFivetranTranslator:
+    """Translator class which converts raw response data from the Fivetran API into AssetSpecs.
+    Subclass this class to implement custom logic for each type of Fivetran content.
+    """
+
+    def __init__(self, context: FivetranWorkspaceData):
+        self._context = context
+
+    @property
+    def workspace_data(self) -> FivetranWorkspaceData:
+        return self._context
+
+    def get_asset_key(self, data: FivetranContentData) -> AssetKey:
+        if data.content_type == FivetranContentType.CONNECTOR:
```

Comment:
shouldn't we also have get_destination_asset_key?

Reply:
We will leverage the previous translator instead. See updated PR description.